api/v1/managementclusterconnection_types.go

```diff
@@ -26,8 +26,32 @@ type ManagementClusterConnectionSpec struct {
 	// should be able to access this address. This field is used by managed clusters only.
 	// +optional
 	ManagementClusterAddr string `json:"managementClusterAddr,omitempty"`
+
+	// TLS provides options for configuring how Managed Clusters can establish an mTLS connection with the Management Cluster.
+	// +optional
+	TLS ManagementClusterTLS `json:"tls,omitempty"`
+}
+
+type ManagementClusterTLS struct {
+	// CA indicates which verification method the tunnel client should use to verify the tunnel server's identity.
+	//
+	// When left blank or set to 'Tigera', the tunnel client will expect a self-signed cert to be included in the certificate bundle
+	//  generated by the Tigera API Server which can be used to verify the tunnel server's identity, and will expect the cert to have
+	// a Common Name (CN) of 'voltron'.
+	//
+	// When set to 'Public', the tunnel client will use its installed system certs and will use the managementClusterAddr to verify the tunnel server's identity.
+	//
+	// Default: Tigera
```

Comment:
I wonder if we are putting too many implementation details in this description.
I think this would do too:

> When left blank or set to 'Tigera', the tunnel client will expect a self-signed cert to be included in the certificate bundle and will verify the Common Name (CN). 


Also, you might need to add validation on this field. Not sure if it is picked up automatically from the type.